core/store/src/flat/chunk_view.rs

```diff
@@ -0,0 +1,46 @@
+use near_primitives::hash::CryptoHash;
+use near_primitives::state::ValueRef;
+
+use crate::Store;
+
+use super::FlatStorage;
+
+/// Struct for getting value references from the flat storage, corresponding
+/// to some block defined in `blocks_to_head`.
+///
+/// The main interface is the `get_ref` method, which is called in `Trie::get`
+/// and `Trie::get_ref` because they are the same for each shard and they are
+/// requested only once during applying chunk.
+// TODO (#7327): lock flat state when `get_ref` is called or head is being updated. Otherwise, `apply_chunks` and
+// `postprocess_block` parallel execution may corrupt the state.
+#[derive(Clone)]
+pub struct FlatStorageChunkView {
+    /// Used to access flat state stored at the head of flat storage.
+    /// It should store all trie keys and values/value refs for the state on top of
+    /// flat_storage.head, except for delayed receipt keys.
+    #[allow(unused)]
+    store: Store,
+    /// The block for which key-value pairs of its state will be retrieved. The flat state
+    /// will reflect the state AFTER the block is applied.
+    block_hash: CryptoHash,
+    /// Stores the state of the flat storage, for example, where the head is at and which
+    /// blocks' state are stored in flat storage.
+    #[allow(unused)]
```

Comment:
This field is used in `get_ref`. I think the general direction of these topics is quite clear – an audit of these `#[allow(unused)]`s is necessary, as some fields are in fact being unconditionally used. And others where the field is unused (but needs to be stored for some other reason), or is used conditionally, `_name` naming scheme and/or `#[cfg(...)]` are more appropriate..